master: nightly linux failed

The nightly linux test run failed on 70aa8f70b9bc3a7fc478585a1513692890d8f694. Please review the run [here](https://github.com/cockroachdb/pebble/actions/runs/15847479863).

```
--- FAIL: TestCompaction (5.31s)
    compaction_test.go:87[6](https://github.com/cockroachdb/pebble/actions/runs/15847479863/job/44672898881#step:4:7): seed: 1750759573776686418
    --- FAIL: TestCompaction/set_with_del_sstable_Pebblev5 (0.09s)
        compaction_test.go:953: 
            testdata/compaction/set_with_del_sstable_Pebblev5:651:
              async-compact a-b L3
            
            expected:
              manual compaction did not block for ongoing
              L4:
                0000[12](https://github.com/cockroachdb/pebble/actions/runs/15847479863/job/44672898881#step:4:13):[a#0,SET-a#0,SET]
                0000[13](https://github.com/cockroachdb/pebble/actions/runs/15847479863/job/44672898881#step:4:14):[b#0,SET-b#0,SET]
            
            found:
              manual compaction blocked until ongoing finished
              L4:
                000012:[a#0,SET-a#0,SET]
                000013:[b#0,SET-b#0,SET]
```

--- a/compaction_test.go
+++ b/compaction_test.go
@@ -1113,18 +1113,12 @@ func TestCompaction(t *testing.T) {
 					close(ch)
 				}()
 
-				manualDone := func() bool {
+				// Wait until the manual compaction is queued.
+				err := try(100*time.Microsecond, 20*time.Second, func() error {
 					select {
 					case <-ch:
-						return true
+						td.Fatalf(t, "manual compaction did not block for ongoing\n%s", s)
 					default:
-						return false
-					}
-				}
-
-				err := try(100*time.Microsecond, 20*time.Second, func() error {
-					if manualDone() {
-						return nil
 					}
 
 					d.mu.Lock()
@@ -1138,8 +1132,11 @@ func TestCompaction(t *testing.T) {
 					return err.Error()
 				}
 
-				if manualDone() {
-					return "manual compaction did not block for ongoing\n" + s
+				// Make sure the manual compaction doesn't complete.
+				select {
+				case <-ch:
+					td.Fatalf(t, "manual compaction did not block for ongoing\n%s", s)
+				case <-time.After(10 * time.Millisecond):
 				}
 
 				d.mu.Lock()
@@ -1216,7 +1213,7 @@ func TestCompaction(t *testing.T) {
 					return err.Error()
 				}
 				if compDone {
-					return "manual compaction did not block for ongoing\n" + s
+					td.Fatalf(t, "manual compaction did not block for ongoing\n%s", s)
 				}
 				// Cancel the manual compaction.
 				(*cancelFunc.Load())()

--- a/testdata/compaction/set_with_del_sstable_Pebblev4
+++ b/testdata/compaction/set_with_del_sstable_Pebblev4
@@ -646,9 +646,8 @@ add-ongoing-compaction startLevel=0 outputLevel=1 start=a end=z
 set-concurrent-compactions max=2
 ----
 
-async-compact a-b L3
+compact a-b L3
 ----
-manual compaction did not block for ongoing
 L4:
   000012:[a#0,SET-a#0,SET]
   000013:[b#0,SET-b#0,SET]

--- a/testdata/compaction/set_with_del_sstable_Pebblev5
+++ b/testdata/compaction/set_with_del_sstable_Pebblev5
@@ -648,9 +648,8 @@ add-ongoing-compaction startLevel=0 outputLevel=1 start=a end=z
 set-concurrent-compactions max=2
 ----
 
-async-compact a-b L3
+compact a-b L3
 ----
-manual compaction did not block for ongoing
 L4:
   000012:[a#0,SET-a#0,SET]
   000013:[b#0,SET-b#0,SET]

--- a/testdata/compaction/set_with_del_sstable_Pebblev6
+++ b/testdata/compaction/set_with_del_sstable_Pebblev6
@@ -648,9 +648,8 @@ add-ongoing-compaction startLevel=0 outputLevel=1 start=a end=z
 set-concurrent-compactions max=2
 ----
 
-async-compact a-b L3
+compact a-b L3
 ----
-manual compaction did not block for ongoing
 L4:
   000012:[a#0,SET-a#0,SET]
   000013:[b#0,SET-b#0,SET]

--- a/testdata/compaction/set_with_del_sstable_Pebblev7
+++ b/testdata/compaction/set_with_del_sstable_Pebblev7
@@ -648,9 +648,8 @@ add-ongoing-compaction startLevel=0 outputLevel=1 start=a end=z
 set-concurrent-compactions max=2
 ----
 
-async-compact a-b L3
+compact a-b L3
 ----
-manual compaction did not block for ongoing
 L4:
   000012:[a#0,SET-a#0,SET]
   000013:[b#0,SET-b#0,SET]

